src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1234,6 +1437,85 @@ int fx_muxer_t::execute(
     return result;
 }
 
+int fx_muxer_t::get_coreclr_delegate(
```

Comment:
This method is doing some really heavy lifting about resolving hostfx, initializing coreclr, etc.  Where alternative smaller building blocks considered to not have them all in this entry point?

Reply:
You have found the only other entry point that should be needed. Merging the app/exe, which is done, and the new lib method might be beneficial but in this case I don't see how it would benefit much. More than willing to reconsider but it would require yet more refactoring that doesn't give as much of a win as other changes.